client/js/utils/api.js

```diff
@@ -8,25 +8,28 @@ var _cache = {};
 var api = {
 
     getAllParks() {
-        var url = '/data/parks.json';
+        var url = 'data/parks.json';
 
         if (_cache[url]) {
             return when.resolve(_cache[url]);
         }
 
-        return ajax({uri: url})
-            .tap((body) => _cache[url] = body);
+        return ajax({url: url})
+            .tap((data) => console.log(data))
```

Comment:
Need to sit down with paper and prove it, but it smells like `Promise` forms a Monad with `tap` as `join` :)